sympy/solvers/solveset.py

```diff
@@ -115,7 +115,17 @@ def invert_real(f_x, y, x, domain=S.Reals):
     Inverts a real-valued function. Same as _invert, but sets
     the domain to ``S.Reals`` before inverting.
     """
-    return _invert(f_x, y, x, domain)
+    lhs, rhs_s = _invert(f_x, y, x, domain)
+    if lhs == x and isinstance(rhs_s, FiniteSet):
+        for rhs in rhs_s:
+            try:
+                if checksol(f_x - y, x, rhs):
+                    pass
+                else:
+                    return (f_x, FiniteSet(y))
```

Comment:
Yes, I guess inverting will always get the desired `solutions`, `checksol` will check each solution, if any `solution` is not passed, it would rather mean that inverting has not been done properly and therefore it would return the system as it is.

Like, in our case `0**x - 100`, the inversion was fine, the reason for failure was during inversion the equation would get to `log(100)/log(0)`, which evaluates to `0`, and therefore this would become the result(which is wrong).